embedded/store/key_reader.go

```diff
@@ -169,7 +167,6 @@ func (s *Snapshot) NewKeyReader(spec *KeyReaderSpec) (*KeyReader, error) {
 		filters:        spec.Filters,
 		refInterceptor: refInterceptor,
 		offset:         spec.Offset,
-		_tx:            s.st.NewTxHolder(),
```

Comment:
I wouldn't expose the txHolder into the exposed APIs but internally fetch the holder from the pool...

Reply:
But that would mean that we'll take the TxHolder for the pool that was meant to limit the commit concurrency. Is that something we're OK with? Thsi may lead to a situation where simultaneous readers could penalize / starve committers.

Reply:
Tx Holder is no longer needed in key reader